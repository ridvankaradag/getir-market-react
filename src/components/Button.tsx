import styled from "styled-components";

interface ButtonProps {
  block?: boolean;
}

const Button = styled.button`
  border: none;
  border-radius: 2px;
  cursor: pointer;
  transition: 0.2s;
`;

export const PrimaryButton = styled(Button)<ButtonProps>`
  background-color: ${(props) => props.theme.colors.primary};
  color: ${(props) => props.theme.colors.white};
  width: ${(props) => (props.block ? "100%" : "auto")};
  &:hover {
    opacity: 0.8;
  }
`;

export default Button;
