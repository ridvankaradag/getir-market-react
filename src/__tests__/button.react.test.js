import React from "react";
import Button, { PrimaryButton } from "../components/Button";

import renderer from "react-test-renderer";

it("Default Button renders correctly", () => {
  const tree = renderer.create(<Button>Default Button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
