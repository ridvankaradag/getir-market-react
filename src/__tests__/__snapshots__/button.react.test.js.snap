// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button renders correctly 1`] = `
<button
  className="sc-bdvvtL hNKPDz"
>
  Default Button
</button>
`;
